kythe/go/extractors/config/smoke/smoke.go

```diff
@@ -38,46 +38,77 @@ import (
 // automatically. The current implementation is to simply check the fraction of
 // files covered by extractor output.
 //
-// TODO(danielmoy): hook up an indexing step and check actual semantic output
-// instead of simple files.
+// Also optionally supports indexing the extractors output.
 //
 // TODO(danielmoy): support more than just java.
-type Tester interface {
-	TestRepo(ctx context.Context, repo string) (Result, error)
-}
+type Tester func(ctx context.Context, repo string) (Result, error)
 
 // Fetcher is a thin wrapper over something which fetches a given repo and
```

Comment:
s/thin wrapper over something which//

```
// A Fetcher fetches the contents of the given repo and writes it to an
// output directory. The implementation may ignore the repo.ConfigPath.

```

Reply:
Not sure if the comment about `ConfigPath` is meant to be a constraint on the implementation, or a description of the current behaviour.

Reply:
yea probably can ommit the later bit.  Really I shouldn't be re-using the same config.Repo since the interface is only "mostly what I want", but mostly just doing it for expediency.

Reply:
:)